### ISSUE TITLE
Language detection fail

### ISSUE THREAD

**Issue by Qronikarz** [ValveSoftware/Proton#6880]
Hello.
In Proton 7.x some games were automatically detecting my OS language correctly - I use Polish language OS with English Steam.

Now when using Proton 8.x the games don't detect it and just use their default. I tried setting Polish language on Steam but it didn't help.

The fastest game to test it is Worms:Armageddon I think, since the default quickmatch just uses detected language for flag and voiceover.

Some more info just in case, but shouldn't matter:
Linux Mint 21
Steam Flatpak version

**Comment by Qronikarz** [comment]
Still the same problem with Proton 9
Switching back to Proton 7.0-6 makes the game detect the language correctly.

**Comment by Qronikarz** [comment]
And once again same thing happens, but I switched OS.

Still the same problem with Proton 10 and switching to 7.0-6 works.

OS: Solus 4.7 KDE Plasma (up-to-date),
Steam flatpak package.

**Comment by alasky17** [comment]
@Qronikarz Could you try adding this as a launch option: ` HOST_LC_ALL=ja_JP.UTF-8 %command% `(except replace ja_JP.UTF-8 with the Polish locale that you use for your OS)

And could you also attach/paste the output of "locale" on your OS?

**Comment by Qronikarz** [comment]
> Could you try adding this as a launch option: `HOST_LC_ALL=ja_JP.UTF-8 %command%`(except replace ja_JP.UTF-8 with the Polish locale that you use for your OS)

Yes, this works. Replaced it with "pl_PL"

> And could you also attach/paste the output of "locale" on your OS?

```
qronikarz@qronikarz-desktop1 ~ $ locale
LANG=pl_PL.utf8
LC_CTYPE="pl_PL.utf8"
LC_NUMERIC=pl_PL.utf8
LC_TIME=pl_PL.utf8
LC_COLLATE="pl_PL.utf8"
LC_MONETARY=pl_PL.utf8
LC_MESSAGES="pl_PL.utf8"
LC_PAPER=pl_PL.utf8
LC_NAME=pl_PL.utf8
LC_ADDRESS=pl_PL.utf8
LC_TELEPHONE=pl_PL.utf8
LC_MEASUREMENT=pl_PL.utf8
LC_IDENTIFICATION=pl_PL.utf8
LC_ALL=
```

**Comment by Qronikarz** [comment]
Tested the Proton 11 beta and it is still not fixed.

**Comment by alasky17** [comment]
@Qronikarz This should be fixed in Proton - Experimental.  Worms:Armageddon is detecting the system language in a very unique way, so we just added a hack for the single game.  If you are experiencing the same regression with other games, please send a full list so we can investigate them individually :)

**Comment by Qronikarz** [comment]
Yes. Can confirm the regression is fixed for Worms:Armageddon. Thank you very much.

I agree that it is detecting the language in interesting way and I don't recall any other games that had such behaviour. I'll close this issue for now and if I ever stumble upon something that I knew it had the translation, but it doesn't set it, then I'll open a new topic.

Have a nice day/night :)
